app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -240,6 +245,25 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
                                     .show()
                             }
 
+                            if (category.isVisible && category.isEnabled){
+                                val oldPasswordDirectory = File("$repoPath/${oldCategory!!.trim('/')}")
```

Comment:
Can you not set the value of this variable right when `oldCategory` is assigned above? Would save an extra variable and probably also the `!!`.